plugins/wporg-5ftf/includes/contributor.php

```diff
@@ -566,3 +583,199 @@ function parse_contributors( $contributors, $pledge_id = null ) {
 
 	return $sanitized_contributors;
 }
+
+/**
+ * Send an email to inactive contributors.
+ */
+function notify_inactive_contributors() : void {
+	$contributors = get_inactive_contributor_batch();
+	$contributors = prune_unnotifiable_xprofiles( $contributors );
+	$contributors = add_user_data_to_xprofile( $contributors );
+	$contributors = prune_unnotifiable_users( $contributors );
+
+	// Limit to 25 emails per cron run, to avoid triggering spam filters.
+	if ( count( $contributors ) > 25 ) {
+		$contributors = array_slice( $contributors, 0, 25 );
+	}
+
+	foreach ( $contributors as $contributor ) {
+		notify_inactive_contributor( $contributor );
+	}
+}
+
+/**
+ * Get the next group of inactive contributors.
+ */
+function get_inactive_contributor_batch() : array {
+	global $wpdb;
+
+	$batch_size = 500; // This can be large because most users will be pruned later on.
+	$offset     = absint( get_option( '5ftf_inactive_contributors_offset', 0 ) );
+
+	$user_xprofiles = $wpdb->get_results( $wpdb->prepare( '
+		SELECT user_id, GROUP_CONCAT( field_id ) AS field_ids, GROUP_CONCAT( value ) AS field_values
+		FROM `bpmain_bp_xprofile_data`
+		WHERE field_id IN ( %d, %d )
+		GROUP BY user_id
+		ORDER BY user_id ASC
+		LIMIT %d
+		OFFSET %d',
+		XProfile\FIELD_IDS['hours_per_week'],
+		XProfile\FIELD_IDS['team_names'],
+		$batch_size,
+		$offset
+	) );
+
+	if ( $user_xprofiles ) {
+		// We haven't reached the end of the totals rows yet.
+		update_option( '5ftf_inactive_contributors_offset', $offset + $batch_size, false );
+
+	} else {
+		// We're at the end of total rows with 0 remainder, so reset.
+		delete_option( '5ftf_inactive_contributors_offset' );
+		return array();
+	}
+
+	$field_names = array_flip( XProfile\FIELD_IDS );
+
+	foreach ( $user_xprofiles as $user ) {
+		$user->user_id = absint( $user->user_id );
```

Comment:
`absint( $user->user_id )` makes me wonder what edge case this is intended for, and whether there's a better way to handle it.

Reply:
IIRC it's just because the DB returns it as a string, but in other places we need to compare it to an int. We have to use strict comparison because the linter will complain if we don't. It also just makes me feel more comfortable, knowing that it's what I expect.